pkg/azure/client/storageaccount.go

```diff
@@ -52,21 +60,30 @@ func (c *StorageAccountClient) CreateStorageAccount(ctx context.Context, resourc
 	return err
 }
 
-// ListStorageAccountKey lists the first key of a storage account.
-func (c *StorageAccountClient) ListStorageAccountKey(ctx context.Context, resourceGroupName, storageAccountName string) (string, error) {
-	response, err := c.client.ListKeys(ctx, resourceGroupName, storageAccountName, &armstorage.AccountsClientListKeysOptions{
-		// doc: "Specifies type of the key to be listed. Possible value is kerb.. Specifying any value will set the value to kerb."
-		Expand: ptr.To("kerb"),
-	})
+// ListStorageAccountKeys lists all keys for the specified storage account.
+func (c *StorageAccountClient) ListStorageAccountKeys(ctx context.Context, resourceGroupName, storageAccountName string) ([]*armstorage.AccountKey, error) {
+	response, err := c.client.ListKeys(ctx, resourceGroupName, storageAccountName, &armstorage.AccountsClientListKeysOptions{})
 
 	if err != nil {
-		return "", err
+		return nil, err
 	}
 
-	if len(response.Keys) < 1 {
-		return "", fmt.Errorf("no key found in storage account %s", storageAccountName)
+	return response.Keys, nil
+}
+
+// RotateKey rotates the key with the given name and returns the updated key.
+func (c *StorageAccountClient) RotateKey(ctx context.Context, resourceGroupName, storageAccountName, storageAccountKeyName string) ([]*armstorage.AccountKey, error) {
+	resp, err := c.client.RegenerateKey(
+		ctx,
+		resourceGroupName,
+		storageAccountName,
+		armstorage.AccountRegenerateKeyParameters{KeyName: ptr.To(storageAccountKeyName)},
+		nil,
+	)
+
+	if err != nil {
+		return nil, err
 	}
 
-	firstKey := response.Keys[0]
-	return *firstKey.Value, nil
+	return resp.Keys, nil
```

Comment:
Nit: If the client methods always return the sorted key slice, there is no need to explicitly sort the keys elsewhere in the code, making it easier to read.
However, if you feel explicitly sorting the keys specifies intent, then we can keep the code as is.

```suggestion
	return SortKeysByAge(resp.Keys), nil
```